File: trees/verValoresEnRango/verValoresEnRango.pas
// Definimos un procedimiento llamado "verValoresEnRango" que toma un puntero a un nodo de árbol (de tipo "arbol"),
// así como dos valores enteros "min" y "max" que representan el rango deseado.
procedure verValoresEnRango(nodo: arbol; min, max: integer);
begin
    // Si el nodo no es nulo (es decir, existe), realizamos lo siguiente:
    if (nodo <> nil) then
    begin
        // Si el valor del nodo actual está dentro del rango [min, max], lo imprimimos
        if (nodo^.valor >= min) and (nodo^.valor <= max) then
        begin
            writeln(nodo^.valor);   
            // Luego, llamamos recursivamente a "verValoresEnRango" para el subárbol izquierdo
            verValoresEnRango(nodo^.hi, min, max);
            // Finalmente, llamamos recursivamente a "verValoresEnRango" para el subárbol derecho
            verValoresEnRango(nodo^.hd, min, max)
        end;
        else if (nodo^.valor > max) then
            verValoresEnRango(nodo^.hi, min, max);
        else if (nodo^.valor < min) then
            verValoresEnRango(nodo^.hd, min, max);
    end;
end;

File: trees/verValoresEnRango/verValoresEnRango.ts
// Importamos el tipo Arbol desde su archivo correspondiente
import { Arbol } from "../type";

// Definimos una función llamada "verValoresEnRango" que toma un nodo (de tipo Arbol o nulo),
// así como dos valores numéricos "min" y "max" que representan el rango deseado.
const verValoresEnRango = (nodo: Arbol | null, min: number, max: number): void => {
  // Si el nodo no es nulo, realizamos lo siguiente:
  if (nodo) {
    // Si el valor del nodo actual está dentro del rango [min, max], lo imprimimos
    if (nodo.valor >= min && nodo.valor <= max) 
      console.log(nodo.valor);
    // Luego, llamamos recursivamente a "verValoresEnRango" para el subárbol izquierdo
    verValoresEnRango(nodo.izquierdo, min, max);
    // Finalmente, llamamos recursivamente a "verValoresEnRango" para el subárbol derecho
    verValoresEnRango(nodo.derecho, min, max);
  }
};
